tests/validations/mta/mtaDiagnostic.spec.ts

```diff
@@ -0,0 +1,165 @@
+      const diagnosticsName = "diagnosticsName";
+      const diagnosticCollection = {};
+      const createDiagnosticCollectionSpy = sinon
+        .stub(testVscode.languages, "createDiagnosticCollection")
+        .returns(diagnosticCollection);
+
+      // cache is empty
+      expect(keys(diagnosticCollections).length).to.equal(0);
+
+      // first call should create the diagnosticCollection
+      const firstCallDc = getDiagnosticsCollection(
+        diagnosticsName,
+        disposables
+      );
+
+      // make sure the diagnosticCollection was created and not cached
+      expect(createDiagnosticCollectionSpy.callCount).to.equal(1);
+      // cache is contain 1 diagnosticCollection
+      expect(keys(diagnosticCollections).length).to.equal(1);
+
+      // second call will should get the diagnosticCollection from the cache
+      const secondCallDc = getDiagnosticsCollection(
+        diagnosticsName,
+        disposables
+      );
+
+      // reference comparison. make sure it is the same object
+      expect(diagnosticCollection === secondCallDc).to.be.true;
+      expect(firstCallDc === secondCallDc).to.be.true;
+      // cache is contain 1 diagnosticCollection
+      expect(keys(diagnosticCollections).length).to.equal(1);
+    });
+
+    it("clearDiagnosticCollections clears the collection", async () => {
+      let clearTriggered = false;
+      const diagnosticCollection = {
+        clear: () => {
+          clearTriggered = true;
+        },
+      };
+      const createDiagnosticCollectionSpy = sinon
+        .stub(testVscode.languages, "createDiagnosticCollection")
+        .returns(diagnosticCollection);
+
+      // cache is empty
+      expect(keys(diagnosticCollections).length).to.equal(0);
+
+      // call should create a new diagnosticCollection
+      getDiagnosticsCollection("diagnosticsName1", disposables);
+
+      expect(keys(diagnosticCollections).length).to.equal(1);
+
+      clearDiagnosticCollections();
+
+      expect(clearTriggered).to.be.true;
+      expect(createDiagnosticCollectionSpy.callCount).to.equal(1);
+    });
+  });
+
+  describe("getDiagnosticsCollection", async () => {
+    it("returns the correct range when column and line are zero", async () => {
+      const mtaIssue: mta.Issue = {
+        severity: "warning",
+        message: "",
+        line: 0,
+        column: 0,
+      };
+
+      const pos = {
+        line: 0,
+        character: 0,
+      };
+      testGetDiagnosticsCollection(pos, mtaIssue);
+    });
+
+    it("returns the correct range when column is zero and line is 2", async () => {
+      const mtaIssue: mta.Issue = {
+        severity: "warning",
+        message: "",
+        line: 2,
+        column: 0,
+      };
+
+      const pos = {
+        line: 1,
+        character: 0,
+      };
+      testGetDiagnosticsCollection(pos, mtaIssue);
+    });
+
+    it("returns the correct range when column and line are 3", async () => {
```

Comment:
Same - what matters here is that the column is non-zero
```suggestion
    it("returns the correct range when column and line non-zero", async () => {
```

Reply:
done